misc/debug-adapter/modules/debug-adapter-core/src/main/java/org/ballerinalang/debugadapter/ExecutionContext.java

```diff
@@ -39,14 +40,16 @@ public class ExecutionContext {
     private String sourceProjectRoot;
     private final DebugProjectCache projectCache;
     private Process launchedProcess;
-    private DebugInstruction lastInstruction;
     private boolean terminateRequestReceived;
     private boolean supportsRunInTerminalRequest;
+    private DebugInstruction prevInstruction;
+    private Location prevLocation;
```

Comment:
Can't we use an optional here ?
```java
private Optional<Location> prevLocation; 
```

Reply:
+1
Updated in https://github.com/ballerina-platform/ballerina-lang/pull/43548/commits/a8bc7cb476ea008469aa9a8018982d9c52f110df.